woeip/apps/data_collection/models.py

```diff
@@ -0,0 +1,118 @@
+
+
+class Organization(models.Model):
+    """An organization, e.g., WOEIP, OpenOakland"""
+    name = models.CharField(max_length=256)
+    website = models.CharField(max_length=256)
+    email = models.CharField(max_length=256)
+    contact = models.ForeignKey('Participant',
+                                related_name='contact', related_query_name='contact',
+                                on_delete=models.SET_NULL, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.name} <{self.email}>"
+
+
+class RoleEnum(models.Model):
+    """Create, read, update, delete permissions"""
+    name = models.CharField(max_length=256)
+    can_create = models.BooleanField()
+    can_read = models.BooleanField()
+    can_update = models.BooleanField()
+    can_delete = models.BooleanField()
+
+    def __str__(self):
+        crud = []
+        if self.can_create:
+            crud.append('create')
+        if self.can_read:
+            crud.append('read')
+        if self.can_update:
+            crud.append('update')
+        if self.can_delete:
+            crud.append('delete')
+
+        crud = '/'.join(crud)
+
+        return f"{self.name} ({crud})"
+
+
+class Participant(models.Model):
+    name = models.CharField(max_length=256)
+    email = models.CharField(max_length=256)
+    organization = models.ForeignKey(Organization, on_delete=models.SET_NULL, blank=True, null=True)
+    role = models.ForeignKey(RoleEnum, on_delete=models.SET_NULL, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.name} <{self.email}> {self.organization} {self.role}"
+
+
+class RouteEnum(models.Model):
+    """Named routes"""
+    name = models.CharField(max_length=256)
+    latlon = LineStringField()
+
+    def __str__(self):
+        return self.name
+
+
+class Session(models.Model):
+    """A single air quality outing. Can link to several SessionData, e.g., raw data files."""
+    date_collected = models.DateTimeField(auto_now_add=True)
+    route = models.ForeignKey(RouteEnum, on_delete=models.SET_NULL, blank=True, null=True)
+    collected_by = models.ForeignKey(Participant, on_delete=models.CASCADE)
+    metadata = JSONField(blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.date_collected} {self.collected_by}"
+
+
+class Device(models.Model):
+    name = models.CharField(max_length=256)
+    manufacturer = models.CharField(max_length=256)
+    serial_number = models.CharField(max_length=256)
+    model_number = models.CharField(max_length=256)
+    calibration_date = models.DateField()
+    firmware_version = models.CharField(max_length=256)
+
+    def __str__(self):
+        return f"{self.name} {self.model_number} {self.serial_number}"
+
+
+class Sensor(models.Model):
+    """A sensor is something that measures something, i.e., it produces a single measurement value
+    at a time.
+    """
+    name = models.CharField(max_length=256)
+    unit = models.CharField(max_length=256,
```

Comment:
How many different units do you expect? If the number is limited, it might make sense to use [`choices`](https://docs.djangoproject.com/en/2.1/ref/models/fields/#choices) here.

Reply:
Eh, I'm no expert, but I've seen dozens of units for different types of sensors. With `choices`, how easy would it be to add units in the future if we forgot a few?

Reply:
Adding a new choice is as simple as adding an item to an array.